112 add generating spreadsheet file

trojczak opened the pull request:
:)

mereolog reviewed (changes requested):
When I ran this PR in docker, search functionality does not work, i.e., every search request (http://localhost:6201/fibo/ontology/api/hint/) ends up 'Failed to load response data: No data found for resource with given identifier.

The main branch 'develop' does not exhibit this problem.

trojczak commented:
@mereolog I can't replicate this problem. Could you check it again with the newest version of the code?

mereolog commented:
It seems to work now, although one needs to wait for quite a time for the service to "warm up".

But I spotted another issue.
The text blocks on the page are displayed not in the same order as in the current prod - note the difference in the location of 'Glossary section'.

**PR-version**
![image](https://user-images.githubusercontent.com/11171688/141753556-feb07080-669b-4f2f-a956-771f9351a0dc.png)

**Prod version**
![image](https://user-images.githubusercontent.com/11171688/141753645-a7173f3d-cc11-401f-babc-c91a16e54f86.png)

(I am running this against the laster master of html-pages.)

mereolog reviewed (changes requested):
There is another issue, which was there for some time, I think.

Logical description 'is denominated in exactly 1 currency' gets translated as 'is denoat leastated in exactly 1 currency'. I think that reason is that 'min' string is replaced with 'at least' string.
Use ' min ' replacement instead?

mereolog commented:
Also the performance is worrying. Running publication of all products took more than 70 minutes on my Mac while the average time of the current time is below 50 minutes.

Perhaps there is nothing we can improve upon here, so this is not a blocker, but 'every little helps'.

trojczak commented:
@mereolog I've fixed the problem with `min` (and also `max`). I've also fixed the problem with the ordering of sections on the page (but can't test it locally as it had already worked for me).

I agree that it takes a lot of time but I think that before we add more changes to the code base, it would be better to add more tests. I also don't want this PR to be open longer than needed.

mereolog commented:
The following two descriptions, marked in red, generated from https://spec.edmcouncil.org/fibo/ontology/FND/Agreements/Contracts/ContractualCommitment are obsolete (and ill-formed, for that matter):

![image](https://user-images.githubusercontent.com/11171688/141800565-251616f4-b421-481e-b506-dab187c44ba8.png)

mereolog reviewed (changes requested):
One restriction in https://spec.edmcouncil.org/fibo/ontology/DER/DerivativesContracts/Swaps/SwapPayingParty is ill-transformed:

![image](https://user-images.githubusercontent.com/11171688/141801858-98cee70e-af7f-4581-bbce-f90b0e5f41f4.png)

mereolog reviewed (changes requested):
Display of https://spec.edmcouncil.org/fibo/ontology/FND/Agreements/Contracts/ContractualCommitment gives this odd artifact:

![image](https://user-images.githubusercontent.com/11171688/141971401-0828e955-6000-4ed0-846c-e4b56974f6f9.png)

This is present neither in https://spec.edmcouncil.org nor in https://github.com/edmcouncil/html-pages/pull/105, so I suspect this is something added in this PR.

mereolog commented:
https://github.com/edmcouncil/onto-viewer/pull/125#pullrequestreview-807128168 is fixed by commit 5634fcf, but https://github.com/edmcouncil/onto-viewer/pull/125#pullrequestreview-806161149 is still open.

mereolog reviewed (changes requested):
Backend for this PR produces a different search json than backend for [prod](https://spec.edmcouncil.org):

[spec_edmcouncil_org_ContractualRestriction.zip](https://github.com/edmcouncil/onto-viewer/files/7569013/spec_edmcouncil_org_ContractualRestriction.zip)

[pr_ContractualRestriction.zip](https://github.com/edmcouncil/onto-viewer/files/7569044/pr_ContractualRestriction.zip)
Why the difference?
